app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,106 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
```

Comment:
You'll implement set_conditional in the abstract base, the conditional will be delegated to the subclass (as `conditional`? idk a good name) set_conditional can enforce its own contract, assign @conditional, or whatever. That way the subclass only needs to know how to implement conditional.